sdk/go/pulumi/provider.go

```diff
@@ -0,0 +1,792 @@
+type ConstructResult struct {
+	URN   URNInput
+	State Input
+}
+
+func (p *providerServer) unmarshalOptions(label string) plugin.MarshalOptions {
+	return plugin.MarshalOptions{
+		Label:         label,
+		KeepUnknowns:  true,
+		KeepSecrets:   true,
+		KeepResources: true,
+	}
+}
+
+func (p *providerServer) marshalOptions(label string) plugin.MarshalOptions {
+	return plugin.MarshalOptions{
+		Label:         label,
+		KeepUnknowns:  true,
+		KeepSecrets:   p.keepSecrets,
+		KeepResources: p.keepResources,
+	}
+}
+
+// GetSchema returns the JSON-encoded schema for this provider's package.
+func (p *providerServer) GetSchema(ctx context.Context,
+	req *pulumirpc.GetSchemaRequest) (*pulumirpc.GetSchemaResponse, error) {
+	if v := req.GetVersion(); v != 0 {
+		return nil, errors.Errorf("unsupported schema version %d", v)
+	}
+	schema := string(p.args.Schema)
+	if schema == "" {
+		schema = "{}"
+	}
+	return &pulumirpc.GetSchemaResponse{Schema: schema}, nil
+}
+
+// CheckConfig validates the configuration for this provider.
+func (p *providerServer) CheckConfig(ctx context.Context,
+	req *pulumirpc.CheckRequest) (*pulumirpc.CheckResponse, error) {
+	return nil, status.Error(codes.Unimplemented, "CheckConfig is not yet implemented")
+}
+
+// DiffConfig diffs the configuration for this provider.
+func (p *providerServer) DiffConfig(ctx context.Context, req *pulumirpc.DiffRequest) (*pulumirpc.DiffResponse, error) {
+	return nil, status.Error(codes.Unimplemented, "DiffConfig is not yet implemented")
+}
+
+// Configure configures the resource provider with "globals" that control its behavior.
+func (p *providerServer) Configure(ctx context.Context,
+	req *pulumirpc.ConfigureRequest) (*pulumirpc.ConfigureResponse, error) {
+
+	p.keepSecrets = req.GetAcceptSecrets()
+	p.keepResources = req.GetAcceptResources()
+	return &pulumirpc.ConfigureResponse{
+		AcceptSecrets:   true,
+		SupportsPreview: true,
+		AcceptResources: true,
+	}, nil
+}
+
+// StreamInvoke dynamically executes a built-in function in the provider. The result is streamed
+// back as a series of messages.
+func (p *providerServer) StreamInvoke(req *pulumirpc.InvokeRequest,
+	server pulumirpc.ResourceProvider_StreamInvokeServer) error {
+	tok := req.GetTok()
+	return errors.Errorf("unknown StreamInvoke token %q", tok)
+}
+
+// Check validates that the given property bag is valid for a resource of the given type and returns
+// the inputs that should be passed to successive calls to Diff, Create, or Update for this
+// resource. As a rule, the provider inputs returned by a call to Check should preserve the original
+// representation of the properties as present in the program inputs. Though this rule is not
+// required for correctness, violations thereof can negatively impact the end-user experience, as
+// the provider inputs are using for detecting and rendering diffs.
+func (p *providerServer) Check(ctx context.Context, req *pulumirpc.CheckRequest) (*pulumirpc.CheckResponse, error) {
```

Comment:
Note: The implementation of these is based on the implementations in https://github.com/pulumi/pulumi/blob/34eae7e7e3ddae475c52f2e248b1eabb23d62cf7/sdk/go/common/resource/plugin/provider_server.go

Reply:
No longer applicable.